doc/source/development/maintaining.rst

```diff
@@ -345,7 +345,126 @@ The benchmarks are scheduled by Airflow. It has a dashboard for viewing and debu
 Release process
 ---------------
 
-The process for releasing a new version of pandas can be found at https://github.com/pandas-dev/pandas-release
```

Comment:
Any ideas/plans for the old release repo?

Reply:
I'd personally archive it once this is merged, and delete it once we released once or twice without using it.

We may want to move couple of scripts here before that happens, for example to automatically download all the wheels, if we don't automate it in the CI before.